CMakeLists.txt

```diff
@@ -1,6 +1,14 @@
-cmake_minimum_required(VERSION 2.8)
-
-project(FlatBuffers)
+cmake_minimum_required(VERSION 3.0)
+
+# Infer version from git describe
+EXECUTE_PROCESS(
```

Comment:
This PR seems to make a lot of possibly unrelated improvements, like e.g. here getting this version from git. Would it be possible to first make a PR that does the absolute minimum to get the main thing you're trying to accomplish done, and then the rest in follow-ups? This will help a cmake amateur like myself have some confidence that this is not going to break something somewhere